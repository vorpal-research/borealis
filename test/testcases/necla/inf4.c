int isoscles;
int scalene;
int triangle;
int equilateral;


int main(int a, int b, int c){
  isoscles=scalene=triangle=equilateral=0;
  
  if (a > 0 && b > 0 && c > 0 && a < b +c){
    triangle = 1;
  } else {
    triangle=-1;
  }

  if (a >= b){
    if ( b >= a){
      isoscles = 1;
    }
  } 
  
  if (b >= c){
    if (c >= b){
      isoscles =1 ;
    }
  }

  if (a >= b){
    if (b >= c){
      if (c >= a){
	equilateral =1;
      }
    }
  }
  
  if (a + b > c){
    triangle=1;
  
    if (isoscles == 0 || equilateral == 0){
      scalene=1;
    }
  }
  if (triangle == 1){
    if (equilateral == 1){
      ASSERT(isoscles == 1);
      ASSERT(scalene== 0);
    } else {
      if (isoscles == 0){
	ASSERT(scalene==1);
      }
    }
  }
  
  return equilateral+isoscles + triangle+scalene;
}
