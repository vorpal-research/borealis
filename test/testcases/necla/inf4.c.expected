
[
	
	{
		"location" : 
		{
			"filename" : "test/testcases/necla/inf4.c",
			"loc" : 
			{
				"col" : 9,
				"line" : 51
			}
		},
		"type" : "REQ-01"
	}
]
